pkg/networkservice/common/monitor/server.go

```diff
@@ -79,14 +79,22 @@ func (m *monitorServer) MonitorConnections(selector *networkservice.MonitorScope
 
 func (m *monitorServer) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*networkservice.Connection, error) {
 	conn, err := next.Server(ctx).Request(ctx, request)
-	eventConn := conn.Clone()
 	if err == nil {
+		if len(conn.GetPath().PathSegments) == 0 {
+			return nil, errors.New("resp.Path.PathSegments has inccorect lengrh = 0")
+		}
+		eventConn := conn.Clone()
 		m.executor.AsyncExec(func() {
-			m.connections[eventConn.GetId()] = eventConn
+			id := eventConn.GetPath().PathSegments[0].GetId()
+			eventConn.Id = id
+			eventConn.Path.Index = 0
```

Comment:
I'm not sure that we should change index here, probabbly we should store the current index.